arch/cpu/cc2538/ieee-addr.c

```diff
@@ -70,11 +69,16 @@ ieee_addr_cpy_to(uint8_t *dst, uint8_t len)
     if(((uint8_t *)IEEE_ADDR_LOCATION)[3] == oui_ti[0]
        && ((uint8_t *)IEEE_ADDR_LOCATION)[2] == oui_ti[1]
        && ((uint8_t *)IEEE_ADDR_LOCATION)[1] == oui_ti[2]) {
-      for(i = 0; i < len / 2; i++) {
-        dst[i] = ((uint8_t *)IEEE_ADDR_LOCATION)[len / 2 - 1 - i];
-      }
-      for(i = 0; i < len / 2; i++) {
-        dst[i + len / 2] = ((uint8_t *)IEEE_ADDR_LOCATION)[len - 1 - i];
+      if(len == 2) {
```

Comment:
The original code block is meant to work for `len==2` as well, so the `if/else` should not be required. Have you spotted a bug here?

Reply:
Wouldn't be nice to avoid the conditional, i.e., fix the for loops so they work on any address len?

Reply:
I had just uploaded a version, where the loops are fixed ...